docs/dependency_confusion.md

```diff
@@ -1,48 +1,30 @@
-# Problem Statement
-
-In February 2021, a [blog post](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)
-popularized a novel way to inject malicious content into applications. The question was raised as to
-whether using Cachi2 makes this issue worse. This document is an analysis of each package manager
-supported by Cachi2 to address the security concern named *dependency confusion*.
-
 # Dependency Confusion
 
-To paraphrase the meaning of
-*[dependency confusion](https://medium.com/@alex.birsan/dependency-confusion-4a5d60fec610)*:
-a package manager is tricked into installing a dependency from an official, public, repository when
-it should’ve installed it from a custom, often private, repository. For example, a company may host
-its own internal npm registry which hosts internal-only packages. Then, someone on the Internet
-uploads a package of the same name to the official npm repository. When the application is
-installed, the package manager chooses the dependency package from the public repository instead
-of the internal one.
+Dependency confusion is a software supply chain attack that substitutes malicious third-party code for a legitimate internal software dependency.
+There are various approaches to creating this kind of attack vector, including:
 
-## Potential Impact
+- Namespacing – by uploading a malicious software library to a public registry named similarly to a trusted, internally-used library, systems that omit a namespace/URL check may mistakenly pull in the malicious code.
+- DNS Spoofing – by using a technique like DNS spoofing, systems can be directed to pull dependencies from malicious repositories while displaying what looks like legitimate internal URLs/paths.
+- Scripting – by modifying build/install scripts or CI/CD pipeline configurations, systems can be tricked into downloading software dependencies from a malicious source rather than a local repository.
 
-What happens when Cachi2 is tricked into downloading a malicious version of one of your dependencies?
-To the system that runs Cachi2, nothing. Cachi2 never executes your code, or the code of any of your
-dependencies. What about your build?
+Source: <https://www.activestate.com/resources/quick-reads/dependency-confusion/>
 
-Hopefully, you are building your application in an isolated environment (such as a container build
-with no network access) to protect your own system. In the best case scenario, the malicious behaviour
-will manifest at build time and the build will fail. In the worst case scenario, the build will succeed
-but the built product will contain malicious code, which may then affect whoever tries to run it.
+## Dependency Confusion and Cachi2
 
-## Dependency Confusion vs Cachi2
+Cachi2 is designed to manage software dependencies without executing any code from those dependencies.
```

Comment:
cachi2 doesn't manage dependencies, that's a package manager's job, we just download the deps.
Again, stating that:
> without executing any code from those dependencies

there's no guarantee from our side this won't happen, we need to be clear on that it's just best effort since we can miss some internal code paths the package manager takes. So, let's be more safe with:

_cachi2 downloads software dependencies doing its best to prevent any arbitrary code execution that may be part of the dependency itself and that could lead to potentially infecting the resulting product, e.g. running a pre/post install script. This requires cachi2 to effectively reverse engineer a given package manager to understand whether the package manager itself can be fully trusted and relied upon to download a dependency using its native mechanisms or we have to work around it by implementing an abstraction ourselves. The only guarantee we can give is that it always needs to be approached with a best effort mindset end employ further attestation mechanisms to the supply chain pipeline._